test/spec/ui/viewer_spec.js

```diff
@@ -232,6 +232,9 @@ describe('ui.viewer.Viewer', function () {
                 text: "Rabbits with cloaks"
             };
             v.load([ann]);
+            window.confirm = function () {
```

Comment:
I think it'd be best to add a test like this and set it to return `false` for `confirm` and assert that `onDelete` isn't called.

Also, you need to fix this so it doesn't have side effects outside this test, probably using `sinon.sandbox`.

Add a local variable, `var sandbox = null` to the `describe` function that encompasses this set of tests and a `beforeEach` and `afterEach` function that initializes and restores it. You can see how that's done elsewhere in the file.